spring-cloud-dataflow-aggregate-task/src/main/java/org/springframework/cloud/dataflow/aggregate/task/impl/AggregateDataFlowTaskExecutionQueryDao.java

```diff
@@ -143,18 +144,54 @@ public class AggregateDataFlowTaskExecutionQueryDao implements DataflowTaskExecu
 			+ "PARENT_EXECUTION_ID, SCHEMA_TARGET"
 			+ " from AGGREGATE_TASK_EXECUTION";
 
+	private static final String GET_EXECUTION_BY_NAME_COMPLETED_BEFORE_END_TIME = "SELECT TASK_EXECUTION_ID,"
```

Comment:
These statements are 99.9% identical so my DRY alarm is going crazy. That being said, you took from the already existing pattern in this class so we should refactor all at once in a separate effort. The current impl (even before this PR) seemed like we are trying to implement JPA repository and we have these permutations of params and they all just result to a piece of info on the WHERE clause (typically). 

If you want to extract a base query that has everything up to the `WHERE` clause for your newly introduced statements only - that would be fine as well. 

I am going to create an issue to refactor the class as a whole.

Reply:
I refactored all statements to reuse the select `SELECT_CLAUSE` and later `GET_EXECUTIONS`